example/src/main/java/ably/io/tutorial/ExampleActivity.java

```diff
@@ -3,11 +3,33 @@
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
 
+import io.ably.lib.rest.AblyRest;
+import io.ably.lib.types.AblyException;
+
 public class ExampleActivity extends AppCompatActivity {
 
+    private final static String API_KEY = "INSERT-YOU-API-KEY-HERE"; /* Sign up on ably.io to get your API key */
+
+    /* RuntimeException will be thrown if API_KEY will not be set to a proper one */
+    static {
+        if (API_KEY.contains("INSERT")) {
+            throw new RuntimeException("API key is not set, sign up on ably.io to get yours");
+        }
+    }
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_example);
+        try {
+            initAbly();
+        } catch (AblyException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /* Add AblyException to method signature as AblyRest constructor can throw one */
+    private void initAbly() throws AblyException {
+        AblyRest ablyRest = new AblyRest(API_KEY);
```

Comment:
I am not sure we should use the REST library in an Android environment.  It may be more suitable to have a separate Java demo which uses REST, and use the Realtime library in Android always as that is more realistic.  I fear this will just confuse Android devs.